modules/swagger-codegen/src/main/resources/asyncscala/api.mustache

```diff
@@ -40,9 +40,14 @@ class {{classname}}(client: TransportClient, config: SwaggerConfig) extends ApiC
     {{/required}}
     {{/queryParams}}
 
-    {{#headerParams}}
-    headerParams += "{{baseName}}" -> {{paramName}}.toString
-    {{/headerParams}}
+  {{#headerParams}}
+      {{^required}}
+          if ({{paramName}} != null) {{paramName}}.map { v => headerParams += "{{baseName}}" -> v.toString }
+      {{/required}}
+      {{#required}}
+          if ({{paramName}} != null) headerParams += "{{baseName}}" -> {{paramName}}.toString
+      {{/required}}
```

Comment:
I do not know the reason (I was not the one who created the async-scala generator).

`scala` generator (API client) also follows similar convention: Option for non-required and raw type for required.

If Option should be used for both required and non-required parameters, I welcome the (breaking) change as long as it conforms to Scala style guide or best practices.